src/syntax/index.ts

```diff
@@ -0,0 +1,160 @@
+import type { Root } from "@wooorm/starry-night";
+import type { ElementContent } from "hast";
+
+import { createStarryNight, common } from "@wooorm/starry-night";
+import sourceTsx from "@wooorm/starry-night/lang/source.tsx";
+import sourceSvelte from "@wooorm/starry-night/lang/source.svelte.js";
+import sourceSolidity from "@wooorm/starry-night/lang/source.solidity.js";
+import sourceToml from "@wooorm/starry-night/lang/source.toml";
+import sourceErlang from "@wooorm/starry-night/lang/source.erlang.js";
+import sourceDockerfile from "@wooorm/starry-night/lang/source.dockerfile";
+import sourceAsciiDoc from "@wooorm/starry-night/lang/text.html.asciidoc";
+
+export type MaybeHighlighted = Root | undefined;
+type StarryNight = Awaited<ReturnType<typeof createStarryNight>>;
```

Comment:
Could we inline the type and change the variable name?

```diff
diff --git a/src/syntax/index.ts b/src/syntax/index.ts
index 89dad44..d1a9849 100644
--- a/src/syntax/index.ts
+++ b/src/syntax/index.ts
@@ -11,7 +11,6 @@ import sourceDockerfile from "@wooorm/starry-night/lang/source.dockerfile";
 import sourceAsciiDoc from "@wooorm/starry-night/lang/text.html.asciidoc";
 
 export type MaybeHighlighted = Root | undefined;
-type StarryNight = Awaited<ReturnType<typeof createStarryNight>>;
 
 export const grammars = [
   ...common,
@@ -31,17 +30,17 @@ export const grammars = [
   },
 ];
 
-let st: StarryNight;
+let starryNight: Awaited<ReturnType<typeof createStarryNight>>;
 
 export async function highlight(
   content: string,
   grammar: string,
 ): Promise<MaybeHighlighted> {
-  if (st === undefined) {
-    st = await createStarryNight(grammars);
+  if (starryNight === undefined) {
+    starryNight = await createStarryNight(grammars);
   }
-  const scope = st.flagToScope(grammar);
-  return st.highlight(content, scope ?? "text.raw");
+  const scope = starryNight.flagToScope(grammar);
+  return starryNight.highlight(content, scope ?? "text.raw");
 }
 
 export function lineNumbersGutter(tree: Root) {

```

Reply:
Yeah I liked the initial `starryNight` also better than `st`

Reply:
5f2d2bd8